### PR TITLE
Credential vault

### PR DESCRIPTION
**Pull request by cdmayer**
Store credentials between sessions

### CODE REVIEW

**Comment by cdmayer** [comment]
@peternied @D4N14L I just need some eyeballs please :) I am planning on filling out the CredentialVault implementation for Desktop before completing the merge, but this has all the changes other than that (and it will be very similar to the WinStore version, just a different storage API).

File: src/OneDrive.Sdk.Authentication.Common/Caching/CredentialCache.cs [hunk 1]
```diff
@@ -20,8 +20,9 @@ public class CredentialCache
     {
         internal readonly IDictionary<CredentialCacheKey, AccountSession> cacheDictionary =
             new ConcurrentDictionary<CredentialCacheKey, AccountSession>();
+        internal CredentialCacheKey MostRecentlyUsedKey { get; set; }
 
-        private const int CacheVersion = 1;
+        private const int CacheVersion = 2;
```
Comment (peternied) [inline src/OneDrive.Sdk.Authentication.Common/Caching/CredentialCache.cs]: With this version revision, it seems like we should create CredentialCacheV2 which does its the new storage/management around the existing CredentialCache object so if we add a CredentialCacheV3 we don't multiplex the number of conditional checks that need to be done, instead we just create a series of very secure nesting dolls

Comment (cdmayer) [inline src/OneDrive.Sdk.Authentication.Common/Caching/CredentialCache.cs]: What do you mean? There's only one check. If cache is older than current version, discard it. Else, continue. That should never change. Also, given that this is sample code, so I don't think it's necessary to bend over backwards to save a couple caches (after all, it just requires going through the OAuth flow once more).

**Comment by peternied** [comment]
:shipit:

**Pull request merged**